smt/surrogate_models/krg_based.py

```diff
@@ -338,24 +405,53 @@ def _predict_variances(self, x):
 
         # Initialization
         n_eval, n_features_x = x.shape
-        x = (x - self.X_mean) / self.X_std
-        # Get pairwise componentwise L1-distances to the input training set
-        dx = manhattan_distances(x, Y=self.X_norma.copy(), sum_over_features=False)
-        d = self._componentwise_distance(dx)
-
-        # Compute the correlation function
-        r = self._correlation_types[self.options["corr"]](
-            self.optimal_theta, d
-        ).reshape(n_eval, self.nt)
 
-        C = self.optimal_par["C"]
-        rt = linalg.solve_triangular(C, r.T, lower=True)
-
-        u = linalg.solve_triangular(
+        if self.options["corr"] != "gower_corr":
+            x = (x - self.X_mean) / self.X_std
+            # Get pairwise componentwise L1-distances to the input training set
+            dx = manhattan_distances(x, Y=self.X_norma.copy(), sum_over_features=False)
+            d = self._componentwise_distance(dx)
+            r = self._correlation_types[self.options["corr"]](
+                self.optimal_theta, d                                  
+            ).reshape(n_eval, self.nt)
+            C = self.optimal_par["C"]
+            rt = linalg.solve_triangular(C, r.T, lower=True)
+
+            u = linalg.solve_triangular(
             self.optimal_par["G"].T,
             np.dot(self.optimal_par["Ft"].T, rt)
             - self._regression_types[self.options["poly"]](x).T,
         )
+        else:
+            # Compute the correlation function
+            Xt = x
+            x_n_rows, x_n_cols = Xt.shape
+            r = np.exp(-gower_matrix(x,data_y=self.X_train,weight = np.asarray(self.optimal_theta)))
+            n_samples, n_features = x.shape
+                
+            if not isinstance(Xt, np.ndarray): 
```

Comment:
same code as in initialize? Is it useful? If yes should be in a separate function